gammapy/maps/axes.py

```diff
@@ -0,0 +1,145 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import numpy as np
+from astropy.time import Time
+
+
+class TimeAxis:
+    """Class representing a time axis.
+
+    Provides methods for transforming to/from axis and pixel coordinates.
+    A time axis can represent non-contiguous sequences of time intervals.
+
+    Parameters
+    ----------
+    time_min : `~astropy.time.Time`
+        Array of edge time values.
+    time_max : `~astropy.time.Time`
+        Array of edge time values.
+    interp : str
+        Interpolation method used to transform between axis and pixel
+        coordinates.  Valid options are 'log', 'lin', and 'sqrt'.
+    name : str
+        Axis name
+    """
+    node_type = "edges"
+    def __init__(self, time_min, time_max, name="", interp="lin",):
+        self._name = name
+
+        invalid = [type(_).__name__ for _ in (time_min, time_max) if not isinstance(_, Time)]
+        if len(invalid)>0:
+            raise TypeError(f"TimeAxis edges must be Time objects. Got {invalid}")
+
+        if not len(time_min) == len(time_max):
+            raise ValueError("Time min and time max must have the same length.")
+
+        if not (np.all(time_min == time_min.sort()) and np.all(time_max == time_max.sort()) ):
+            raise ValueError("TimeAxis: edge values must be sorted")
+
+        self._time_min = Time(time_min)
+        self._time_max = Time(time_max)
+
+        self._interp = interp
+
+        self._pix_offset = -0.5
+        self._nbin = len(time_min)
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def nbin(self):
+        return len(self._time_min)
+
+    @property
+    def time_min(self):
```

Comment:
Here I'm not 100% sure whether we should introduce the additional properties with names adjusted to the time content. I think for now it;s definitely more important to stay consistent with the `MapAxis` object, because we definitely want to avoid any kind of extra handling...

Reply:
I think the time content are useful to obtain real time objects. We can still add the `edges_min` and `edges_max` properties but they will return quantities relative to the reference time.